oap-server/server-cluster-plugin/cluster-zookeeper-plugin/src/main/java/org/apache/skywalking/oap/server/cluster/plugin/zookeeper/ClusterModuleZookeeperProvider.java

```diff
@@ -61,7 +79,39 @@ public ClusterModuleZookeeperProvider() {
 
     @Override public void prepare() throws ServiceNotProvidedException, ModuleStartException {
         RetryPolicy retryPolicy = new ExponentialBackoffRetry(config.getBaseSleepTimeMs(), config.getMaxRetries());
-        client = CuratorFrameworkFactory.newClient(config.getHostPort(), retryPolicy);
+
+        CuratorFrameworkFactory.Builder builder = CuratorFrameworkFactory.builder()
+            .retryPolicy(retryPolicy)
+            .connectString(config.getHostPort());
+        if (config.isEnableACL()) {
+            final List<ACL> acls = Lists.newArrayList();
+
+            String authInfo = config.getAuth();
+            if ("digest".equals(config.getSchema())) {
```

Comment:
I am not familiar w/ ACL, but why is `digest` schema special?

Reply:
When used in the ACL the expression will be the username:base64 encoded SHA1 password digest. But when used in auth the expression must not encode.

The other schemas, such as `IP`, it doesn't have this problem.

Reply:
Is this the official recommended way? Look like strange that, a project like Zookeeper providing such strange style of API.

Reply:
It's not good. I will try others.

Reply:
The Zookeeper Document said:
> digest uses a username:password string to generate MD5 hash which is then used as an ACL ID identity. Authentication is done by sending the username:password in clear text. When used in the ACL the expression will be the username:base64 encoded SHA1 password digest.

Most projects only provided `digest` schema, such as Solr.